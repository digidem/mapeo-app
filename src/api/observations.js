// @flow
import fetch from 'fetch';
import { jsonRequest } from './base';

export interface CreateRequest {
  lat: number;
  lon: number;
}

export type UpdateRequest = {
  id: string
};

class Observation {
  static list = () =>
    jsonRequest({
      method: 'GET',
      route: '/observations'
    });

  static create = (observation: CreateRequest) =>
    jsonRequest({
      method: 'POST',
      route: '/observations',
      body: {
        device_id: '1',
        ...observation
      }
    });

  static update = (observation: UpdateRequest) =>
    jsonRequest({
      method: 'PUT',
      route: `/observations/${observation.id}`,
      body: {
        ...observation
      }
    });
}

export default Observation;
