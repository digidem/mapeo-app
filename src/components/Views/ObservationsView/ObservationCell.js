// @flow
import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Dimensions,
  Image
} from 'react-native';
import I18n from 'react-native-i18n';
import type { Observation } from '../../../types/observation';
import { LIGHT_GREY } from '../../../lib/styles';
import moment from '../../../lib/localizedMoment';

export type Props = {
  currentLocale: string,
  observation: Observation,
  onPress: (i: Observation) => void
};

const styles = StyleSheet.create({
  container: {
    height: 80,
    flexDirection: 'row',
    backgroundColor: 'white',
    justifyContent: 'flex-start',
    alignItems: 'center',
    borderBottomColor: LIGHT_GREY,
    borderBottomWidth: 1,
    width: Dimensions.get('window').width,
    paddingHorizontal: 20
  },
  header: {
    fontSize: 20,
    textAlign: 'center',
    paddingVertical: 20,
    fontWeight: '700',
    color: 'black'
  },
  text: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'flex-start'
  },
  circle: {
    width: 50,
    height: 50,
    backgroundColor: 'white',
    borderRadius: 50,
    borderColor: LIGHT_GREY,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: -5,
    shadowColor: 'black',
    shadowRadius: 10,
    shadowOpacity: 1,
    shadowOffset: { width: 0, height: 2 },
    elevation: 3
  },
  circleWithMedia: {
    position: 'absolute',
    right: 0,
    bottom: -5,
    width: 25,
    height: 25,
    zIndex: 5
  },
  icon: {
    width: 15,
    height: 15
  },
  iconWithMedia: {
    width: 12,
    height: 12
  },
  innerCircle: {
    width: 40,
    height: 40,
    backgroundColor: LIGHT_GREY,
    borderRadius: 50
  },
  title: {
    fontSize: 18,
    fontWeight: '700',
    color: 'black'
  },
  titleLong: {
    fontSize: 16,
    fontWeight: '700',
    color: 'black'
  },
  media: {
    width: 60,
    height: 60,
    borderRadius: 7
  }
});

I18n.fallbacks = true;
I18n.translations = {
  en: require('../../../translations/en'),
  es: require('../../../translations/es')
};

const ObservationCell = (props: Props) => {
  const esLocale = require('moment/locale/es');
  const currentLocale = props.currentLocale;
  const { observation, onPress } = props;
  let dateString;
  if (currentLocale && currentLocale.includes('es')) {
    dateString = moment(props.observation.created).calendar(null, {
      sameDay: '[Hoy], h:mm A',
      nextDay: '[MaÃ±ana], h:mm A',
      nextWeek: 'ddd, h:mm A',
      lastDay: '[Ayer], h:mm A',
      lastWeek: 'ddd, h:mm A',
      sameElse: 'MM/D/YYYY, h:mm A'
    });
  } else {
    dateString = moment(props.observation.created).calendar(null, {
      sameDay: '[Today], h:mm A',
      nextDay: '[Tomorrow], h:mm A',
      nextWeek: 'ddd, h:mm A',
      lastDay: '[Yesterday], h:mm A',
      lastWeek: 'ddd, h:mm A',
      sameElse: 'MM/D/YYYY, h:mm A'
    });
  }

  const handlePress = () => {
    onPress(observation);
  };

  const hasMedia = observation && !!observation.media.length;

  return (
    <TouchableOpacity onPress={handlePress}>
      <View style={styles.container}>
        <View style={styles.text}>
          <Text style={styles.title}>{dateString}</Text>
          <Text>
            {I18n.t(
              `categories.${
                observation.categoryId ? observation.categoryId : ''
              }`
            )}
          </Text>
        </View>
        <View style={{ flexDirection: 'column' }}>
          {hasMedia && (
            <Image
              source={{ uri: observation.media[0].source }}
              style={styles.media}
            />
          )}
          <View style={[styles.circle, hasMedia ? styles.circleWithMedia : {}]}>
            {!!observation &&
              !!observation.icon && (
                <View
                  style={{
                    alignItems: 'center',
                    justifyContent: 'center'
                  }}
                >
                  <Image
                    source={observation.icon}
                    style={hasMedia ? styles.iconWithMedia : styles.icon}
                    resizeMode="contain"
                  />
                </View>
              )}
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default ObservationCell;
