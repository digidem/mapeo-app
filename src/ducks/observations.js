// @flow

import update from 'immutability-helper';
import {keyBy} from 'lodash';
import {create} from '../lib/redux';
import {createSelector} from '../lib/selector';
import type {StoreState} from '../types/redux';
import type {Observation} from '../types/observation';

export const {
  type: OBSERVATION_LIST,
  action: observationList,
  reducer: observationListReducer
} = create('OBSERVATION_LIST', {
  success: (state, action) => {
    const observations = keyBy(action.payload, 'id');

    if (state.observations) {
      Object.keys(state.observations).forEach(id => observations[id] = {
        ...observations[id],
        ...state.observations[id],
      });
    }

    const newState = update(state, {observations: {$set: observations}});

    return newState;
  }
});

export const {
  type: OBSERVATION_CREATE,
  action: observationCreate,
  reducer: observationCreateReducer
} = create('OBSERVATION_CREATE', {
  start: (state, action) => {
    const observation = action.meta;

    if (!observation.lat && !observation.lon && state.gps.data) {
      observation.lat = state.gps.data.latitude;
      observation.lon = state.gps.data.longitude;
    }

    const newState = update(state, {selectedObservation: {$set: observation}});

    return newState;
  },
  success: (state, action) => {
    const newState = update(state, {selectObservation: {$set: action.payload}});

    return newState;
  }
});

export const {
  type: OBSERVATION_UPDATE,
  action: observationUpdate,
  reducer: observationUpdateReducer
} = create('OBSERVATION_UPDATE', {
  start: (state, action) => {
    let newState;
    if (state.selectedObservation &&
        action.meta.id === state.selectedObservation.id) {
      newState = update(state, {
        selectedObservation: {
          $set: {...state.selectedObservation, ...action.meta},
        }
      });
    } else {
      newState = update(state, {
        observations: {
          [action.meta.id]: {
            $set: {...state.observations[action.meta.id], ...action.meta}
          }
        }
      });
    }

    return newState;
  }
});

export const {
  type: OBSERVATION_SELECT,
  action: observationSelect,
  reducer: observationSelectReducer
} = create('OBSERVATION_SELECT', {
  start: (state, action) => {
    if (action.meta === undefined && state.selectedObservation) {
      return update(state, {
        selectedObservation: {$set: action.meta},
        observations: {
          [state.selectedObservation.id]: {
            $set: {
              ...(state.observations[state.selectedObservation.id] || {}),
              ...state.selectedObservation,
            }
          }
        }
      });
    } else {
      return update(state, {selectedObservation: {$set: action.meta}});
    }
  }
});

export const selectObservation = createSelector(
    [(state: StoreState, id: string): Observation =>
         state.app.observations[id]],
    (observation: Observation): Observation => observation,
    (observation: Observation, id: string): string => id);

export default [
  observationListReducer, observationCreateReducer, observationUpdateReducer,
  observationSelectReducer
];
