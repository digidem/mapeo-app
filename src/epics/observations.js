// @flow
import React from 'react';
import { Image } from 'react-native';
import type { ActionsObservable } from 'redux-observable';
import {
  OBSERVATION_LIST,
  observationList,
  OBSERVATION_CREATE,
  observationCreate,
  OBSERVATION_UPDATE,
  observationUpdate
} from '../ducks/observations';
import { Action } from '../types/redux';
import type { Observation as ObservationType } from '../types/observation';
import type { StoreState } from '../types/redux';
import Observation from '../api/observations';
import type { CreateRequest, UpdateRequest } from '../api/observations';

export const observationListEpic = (
  action$: ActionsObservable<Action<string, ObservationType[]>>,
  store: StoreState
) =>
  action$
    .ofType(OBSERVATION_LIST)
    .filter(action => action.status === 'Start')
    .flatMap(() =>
      Observation.list().map(observations => {
        console.log('RN - Observations List', observations);
        return observationList('', observations);
      })
    );

export const observationCreateEpic = (
  action$: ActionsObservable<Action<CreateRequest, ObservationType>>,
  store: StoreState
) =>
  action$
    .ofType(OBSERVATION_CREATE)
    .filter(action => action.status === 'Start')
    .flatMap(action =>
      Observation.create(action.meta).map(observation => {
        console.log('RN - Observation Created');
        return observationCreate(action.meta, observation);
      })
    );

// export const observationUpdateEpic = (
//   action$: ActionsObservable<Action<UpdateRequest, ObservationType>>,
//   store: StoreState
// ) =>
//   action$
//     .ofType(OBSERVATION_UPDATE)
//     .filter(action => action.status === 'Start')
//     .flatMap(action =>
//       Observation.update(action.meta).map(observation => {
//         console.log('RN - Observation Updated - ', action.meta, observation);
//         return observationUpdate(action.meta, observation);
//       })
//     );

export default [
  observationListEpic,
  observationCreateEpic
  // observationUpdateEpic
];
