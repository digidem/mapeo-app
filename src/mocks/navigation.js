import type { NavigationScreenProp } from 'react-navigation';

export const createNavigationScreenProp = () => {
  return {
    state: {},
    dispatch: jest.fn(),
    addListener: jest.fn(),
    getParam: jest.fn(),
    isFocused: jest.fn(),
    goBack: jest.fn(),
    navigate: jest.fn(),
    setParams: jest.fn()
  };
};
