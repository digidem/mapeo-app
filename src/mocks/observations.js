// @flow

export const createObservation = (observation?: any) => ({
  type: 'Observation',
  id: 1,
  lat: 50.5,
  lon: 50.5,
  link: 'linkID',
  created: new Date('2018-02-25T20:23:00.000Z'),
  name: 'Oil Spill',
  notes:
    'This is pretty bad. It smells pretty noxious, and the entire pond is affected.',
  observedBy: 'You',
  media: [],
  icon: require('../images/categories/category_18.png'),
  categoryId: 19,

  ...observation
});
