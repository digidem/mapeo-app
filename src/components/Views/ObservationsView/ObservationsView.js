// @flow
import React from 'react';
import {
  TouchableOpacity,
  TouchableWithoutFeedback,
  Text,
  View,
  FlatList,
  Dimensions
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import SyncIcon from 'react-native-vector-icons/FontAwesome';
import SettingsIcon from 'react-native-vector-icons/MaterialIcons';
import { orderBy, map } from 'lodash';
import I18n from 'react-native-i18n';
import type { Observation } from '../../../types/observation';
import type { Category } from '../../../types/category';
import ObservationCell from './ObservationCell';
import ObservationHeader from './ObservationHeader';
import moment from '../../../lib/localizedMoment';

export type StateProps = {
  drawerOpened: boolean,
  observations: Observation[],
  categories: {
    [id: string]: Category
  },
  icons: Object
};

export type DispatchProps = {
  selectObservation: (o: Observation) => void,
  listCategories: () => void,
  listObservations: () => void
};

type Props = {
  closeRightDrawer: Function,
  navigation: NavigationActions
};

I18n.fallbacks = true;
I18n.translations = {
  en: require('../../../translations/en'),
  es: require('../../../translations/es')
};

class ObservationsView extends React.Component<
  Props & StateProps & DispatchProps,
  State
> {
  componentDidMount() {
    this.props.listObservations();
    if (!Object.keys(this.props.categories).length) {
      this.props.listCategories();
    }
  }

  shouldComponentUpdate(
    nextProps: Props & StateProps & DispatchProps,
    nextState: State
  ) {
    if (nextProps !== this.props || nextState !== this.state) {
      if (
        nextProps.navigation.isFocused() &&
        !this.props.navigation.isFocused()
      ) {
        nextProps.listObservations();
      }
      return true;
    }
    return false;
  }

  render() {
    const {
      navigation,
      observations,
      categories,
      selectObservation,
      icons
    } = this.props;
    const sectionMappings = {};
    let label;

    const keyExtractor = item => item.id.toString();
    const handleItemPress = item => {
      selectObservation(item);
      navigation.navigate({ routeName: 'ObservationDetailView' });
    };
    const goToSettings = () => {
      navigation.navigate({ routeName: 'SettingsView' });
    };
    const goToSyncView = () => {
      navigation.navigate({ routeName: 'SyncView' });
    };

    return (
      <TouchableWithoutFeedback>
        <View
          style={{
            flex: 1
          }}
        >
          <FlatList
            scrollEnabled
            stickyHeaderIndices={[0]}
            ListHeaderComponent={
              <ObservationHeader
                closeRightDrawer={this.props.closeRightDrawer}
                goToSyncView={goToSyncView}
                onSettingsPress={goToSettings}
              />
            }
            style={{ width: Dimensions.get('window').width }}
            keyExtractor={keyExtractor}
            renderItem={({ item }) => (
              <ObservationCell
                currentLocale={I18n.currentLocale()}
                navigation={this.props.navigation}
                observation={item}
                onPress={handleItemPress}
                category={categories[item.categoryId]}
                icons={icons}
              />
            )}
            data={observations}
          />
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

export default ObservationsView;
