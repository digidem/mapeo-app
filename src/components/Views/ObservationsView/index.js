// @flow
import { connect } from 'react-redux';
import { withNavigationFocus } from 'react-navigation';
import { values } from 'lodash';
import type { Dispatch } from 'redux';
import { StoreState } from '../../../types/redux';

import {
  observationSelect,
  observationList
} from '../../../ducks/observations';
import { categoryList } from '../../../ducks/categories';
import { fieldList } from '../../../ducks/fields';
import ObservationsView from './ObservationsView';
import type { StateProps, DispatchProps } from './ObservationsView';

function mapStateToProps(state: StoreState): StateProps {
  const observations = values(state.app.observations).sort(
    (a, b) => new Date(b.created) - new Date(a.created)
  );
  const drawerOpened = state.app.drawers.observations;
  return {
    drawerOpened,
    observations,
    categories: state.app.categories,
    icons: state.app.icons
  };
}

function mapDispatchToProps(dispatch: Dispatch): DispatchProps {
  return {
    selectObservation: o => dispatch(observationSelect(o)),
    listCategories: () => {
      dispatch(categoryList(''));
      dispatch(fieldList(''));
    },
    listObservations: () => dispatch(observationList(''))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(
  withNavigationFocus(ObservationsView)
);
