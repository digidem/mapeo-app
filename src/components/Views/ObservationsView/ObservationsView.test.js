// @flow
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { NavigationActions } from 'react-navigation';
import ObservationsView from './ObservationsView';
import { createObservation } from '../../../mocks/observations';
import { createNavigationScreenProp } from '../../../mocks/navigation';

jest.mock('./ObservationCell', () => 'mock-observation-cell');

describe('ObservationsView tests', () => {
  const closeRightDrawer = jest.fn();
  const observation = createObservation();
  const cases = [observation];

  beforeEach(() => {
    closeRightDrawer.mockReset();
  });

  test('snapshot', () => {
    const tree = renderer.create(
      <ObservationsView
        drawerOpened={true}
        observations={cases}
        categories={{}}
        icons={{}}
        selectObservation={jest.fn()}
        closeRightDrawer={closeRightDrawer}
        navigation={createNavigationScreenProp()}
        listCategories={jest.fn()}
        listObservations={jest.fn()}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
