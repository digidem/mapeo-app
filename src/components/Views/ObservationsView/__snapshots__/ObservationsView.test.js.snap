// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationsView tests snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
  testID={undefined}
>
  <RCTScrollView
    ListHeaderComponent={
      <ObservationHeader
        closeRightDrawer={[Function]}
        goToSyncView={[Function]}
        onSettingsPress={[Function]}
      />
    }
    data={
      Array [
        Object {
          "categoryId": 19,
          "created": 2018-02-25T20:23:00.000Z,
          "fields": Array [
            Object {
              "answer": "",
              "answered": false,
              "id": "0",
              "name": "fieldName",
              "placeholder": "this is some placeholder text",
              "type": "text",
            },
          ],
          "icon": Object {
            "testUri": "../../../src/images/categories/category_18.png",
          },
          "id": 1,
          "lat": 50.5,
          "link": "linkID",
          "lon": 50.5,
          "media": Array [],
          "name": "Oil Spill",
          "notes": "This is pretty bad. It smells pretty noxious, and the entire pond is affected.",
          "observedBy": "You",
          "type": "Observation",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    style={
      Object {
        "width": 750,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "height": 65,
                "justifyContent": "space-between",
                "paddingHorizontal": 15,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#a5a5a4",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ed6109",
                    "borderRadius": 50,
                    "height": 35,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 35,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "backgroundColor": "#ff9933",
                      "borderRadius": 50,
                      "height": 30,
                      "justifyContent": "center",
                      "width": 30,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#ffffff",
                          "fontSize": 20,
                        },
                        Object {
                          "transform": Array [
                            Object {
                              "rotate": "15deg",
                            },
                          ],
                        },
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#a5a5a4",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#f7f7f7",
                "borderBottomWidth": 1,
                "borderColor": "#EAEAEA",
                "borderTopWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "fontWeight": "700",
                }
              }
            >
              All Observations
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              View by Most Recent
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#EAEAEA",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "flex-start",
                "paddingHorizontal": 20,
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "fontWeight": "700",
                  }
                }
              >
                02/25/2018, 12:23 PM
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                19
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#EAEAEA",
                      "borderRadius": 50,
                      "borderWidth": 1,
                      "elevation": 3,
                      "height": 50,
                      "justifyContent": "center",
                      "marginRight": -5,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 10,
                      "width": 50,
                    },
                    Object {},
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
